web/src/settings_users.ts

```diff
@@ -807,3 +831,42 @@ export function fetch_presence_user_setting({render_table}: FetchPresenceUserSet
         },
     });
 }
+
+export async function website_presence(user_id: number): Promise<string> {
+    const last_active_date = presence.last_active_date(user_id);
+
+    if (last_active_date) {
+        return user_last_seen_time_status(user_id);
+    }
+
+    const result = await user_last_seen(user_id, () =>
+        timerender.last_seen_status_from_date(new Date(aggregate_presence.timestamp * 1000)),
+    );
+    return result;
+}
+
+export const user_last_seen = async (user_id: number, callback: () => string): Promise<string> => {
+    const url = `json/users/${user_id}/presence`;
+
+    return new Promise((resolve) => {
+        channel.get({
+            url,
+            success(data: unknown) {
+                const parsed_data = user_last_seen_response_schema.safeParse(data);
+
+                if (parsed_data.success) {
+                    const response = parsed_data.data;
+
+                    if (response.result === "success" && response.presence) {
+                        const {aggregated} = response.presence;
+
+                        aggregate_presence.status = aggregated.status;
+                        aggregate_presence.timestamp = aggregated.timestamp;
```

Comment:
Can't we do something like calling the function to open the popover here after the presence response is received.

Code calling the code to open user card can see if presence data is available and can open the popover directly if data is available. Otherwise can just call the api to get the data and then the `success` callback can have code to open the card. I am not sure which approach would be better but @timabbott might have some thoughts.

Reply:
Hey @timabbott, could you please share your thoughts on this so I can proceed further?

Reply:
I think the best UX for this situation is to immediately show the popover (with a loading spinner in the relevant location) without waiting for the presence information to be fetched from the server, and once we get that information, replace the loading spinner with the value.

99% of the time that someone opens that popover, they're looking for something other than the last active date for the user, so I think it's appropriate to prioritize the invariant that the popover always opens instantly over having it always be complete.